flatdata-cpp/include/flatdata/MemoryMappedTarFileStorage.h

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2021 HERE Europe B.V.
+ * See the LICENSE file in the root of this project for license details.
+ */
+
+#pragma once
+
+#include "MemoryDescriptor.h"
+#include "internal/TarReader.h"
+
+#include <boost/interprocess/file_mapping.hpp>
+#include <boost/interprocess/mapped_region.hpp>
+
+#include <cstdio>
+#include <map>
+#include <memory>
+
+namespace flatdata
+{
+class MemoryMappedTarFileStorage
+{
+public:
+    explicit MemoryMappedTarFileStorage( const char* tar_path );
+
+    MemoryDescriptor read( const char* path ) const;
+
+private:
+    boost::interprocess::mapped_region m_region;
+    std::map< std::string, MemoryDescriptor > m_files;
+};
+
+inline MemoryMappedTarFileStorage::MemoryMappedTarFileStorage( const char* tar_path )
+{
+    try
+    {
+        boost::interprocess::file_mapping file( tar_path, boost::interprocess::read_only );
+        boost::interprocess::mapped_region region( file, boost::interprocess::read_only );
+        if ( region.get_size( ) == 0 )
+        {
+            return;
+        }
+
+        m_region = std::move( region );
+    }
+    catch ( boost::interprocess::interprocess_exception& )
+    {
+        return;
+    }
+
+    auto files = internal::read_tar_files( tar_path );
```

Comment:
Why not pass the memory mapped region on to the TAR reader, instead of re-reading the data through conventional APIs? `read` would then just be a `memcpy`